codility_training/lessons.lesson17.DynamicProgramming.NumberSolitaire/sjjyy.java

```diff
@@ -0,0 +1,53 @@
+package lessons.lesson17.DynamicProgramming.NumberSolitaire;
+
+import java.util.Arrays;
+
+public class sjjyy
+{
+    public static int solution(int[] A)
+    {
+        int len = A.length;
+        int[] d = new int[len];
+        Arrays.fill(d, Integer.MIN_VALUE);
+        d[0] = A[0];
+
+        for(int i = 0 ; i < len ; i++)
+        {
+            for(int j = i + 1 ; j < len && j < i + 7 ; j++)
+            {
+                int max = d[i] + A[j];
+                d[j] = Math.max(d[j], max);
+            }
+        }
+
+        return d[len-1];
+    }
+
+    public static int sol2(int[] A)
+    {
+        int len = A.length;
+        int[] d = new int[len];
+        d[0] = A[0];
+
+        for(int i = 1 ; i < len ; i++)
+        {
+            int max = Integer.MIN_VALUE;
+
+            for(int j = 0 ; j < 6 ; j++)
+            {
+                if(i>=j)
+                    max = Math.max(d[i-j]+A[i], max);
+            }
+            d[i] = max;
```

Comment:
```suggestion
            d[i] = IntStream(0, 6).reduce(Integer.MIN_VALUE, (max, j) -> {
                if (i >= j) max = Math.max(d[i - j] + A[i], max);
                return max;
            });
```
이렇게 표현할 수 있습니다!

혹은

```
            d[i] = IntStream(0, 6).reduce(
                Integer.MIN_VALUE,
                (max, j) -> i >= j ? Math.max(d[i - j] + A[i], max) : max
            );
```
이렇게도 가능합니다